src/components/search-results/crew/crew.component.js

```diff
@@ -4,14 +4,44 @@ import { NavLink } from "react-router-dom";
 
 import ItemInfo from "../../partials/item-info/item-info";
 
-import { CREW_PAGE } from "../../../root/root.constants.js";
+import { CREW_PAGE, VIEW_CREW_PAGE } from "../../../root/root.constants.js";
 
 import "../search-results.css";
 
+// FIXME: move to get-data and DRY with goCrewViewPage
+export const getVesselViewPageLinkWithFilter = (crew) => {
+  const filter = {};
+  if (crew.isCaptain || crew.rank === "captain") {
+    if (crew.license) {
+      filter["captain.license"] = crew.license;
+    }
+    if (crew.name) {
+      filter["captain.name"] = crew.name;
+    }
+  } else {
+    if (crew.license) {
+      filter["crew.license"] = crew.license;
+    }
+    if (crew.name) {
+      filter["crew.name"] = crew.name;
+    }
+  }
+
+  // FIXME: DRY with goToPageWithFilter
+  return VIEW_CREW_PAGE.replace(":filter", JSON.stringify(filter))
+};
+
 class FoundCrew extends Component {
   render() {
     const { crewList, total, searchWords } = this.props;
 
+    const crew = crewList[0];
+
+    console.log(crew)
+    // TODO: get other attributes from query to formulate proper query for link
+    // license is undefined -> need this from query
+    // { name, rank } is defined
+
```

Comment:
still in progress...

Reply:
the link works, but since I'm not getting the `license` from the search results payload, the resulting link doesn't have the proper query

Reply:
Not every crew memeber has a license (there's even a violation for that!).

We did have a similar issue on the crew member page, that when the license is null it wasn't displaying the name properly. That was issue #322 - and the fix allowed a URL like:
/#/crew/view/{"crew.name":"Stowaway"}
to work properly.

So maybe you could pass that JSON query in the URL?

Reply:
> Not every crew memeber has a license (there's even a violation for that!).

Good point @Sheeri 

However, in the case of Jack Harnass, he has a license, but the Search Service just doesn't seem to get the full details
It only gets the name

https://github.com/WildAid/o-fish-web/blob/main/src/services/search.service.js#L28-L36

The `getRanks` tries to get those details, but the `data.crew` objects returned by `stitchService.client.callFunction("search", [critheria])` 
already don't have the attribuets to begin with...

![image](https://user-images.githubusercontent.com/14609656/97505794-f21c8c00-194f-11eb-8769-ecaf7d04683f.png)


On the other hand `data.boardings` returned by the same search returns a more "complete" object

![image](https://user-images.githubusercontent.com/14609656/97506153-a9b19e00-1950-11eb-9526-ef4272b381ed.png)

Could this be a service issue?
Where can I see the implementation of `stitchService.client.callFunction("search", [critheria])` ?

Reply:
@Sheeri 
I started looking at the MongoDB functions in Realm
and confirmed that the "search" function only returns the names for Crew and Vessels

![image](https://user-images.githubusercontent.com/14609656/97563100-f7fa8780-19b8-11eb-910c-5fa52927f423.png)

...and for the boarding it includes other other stuff

![image](https://user-images.githubusercontent.com/14609656/97563168-0c3e8480-19b9-11eb-842c-9ef6fb0bfec9.png)

Reply:
It seems that the most likely solution is add the `crew.license` here for the crew results, for this function
Can I submit a PR to the ofish-realm?

That would solve this issue
(which is the same issue we discussed in my old PR
https://github.com/WildAid/o-fish-web/pull/327#discussion_r508166445
)

Reply:
Absolutely, please do submit a PR to o-fish-realm!

Reply:
@Sheeri I opened a PR to o-fish-realm
https://github.com/WildAid/o-fish-realm/pull/196